src/query/query_manager.c

```diff
@@ -1391,14 +1391,23 @@ xqmgr_execute_query (THREAD_ENTRY * thread_p, const XASL_ID * xasl_id_p, QUERY_I
 
       if (do_not_cache == false)
 	{
-	  /* lookup the list cache with the parameter values (DB_VALUE array) */
-	  list_cache_entry_p =
-	    qfile_lookup_list_cache_entry (thread_p, xasl_cache_entry_p->list_ht_no, &params, &cached_result);
-	  /* If we've got the cached result, return it. */
-	  if (cached_result)
+	  /* lookup the hash table number for qfile list cache */
+	  if (xasl_cache_entry_p->list_ht_no < 0)
```

Comment:
When ht_no is newly allocated, it is not necessary to call qfile_lookup_list_cache_entry(). This code can be moved back several lines.

Reply:
very good point. thanks.